clusters/build-clusters/01_cluster/openshift/api_audit_log/admin_fluentd-configmap.yaml

```diff
@@ -0,0 +1,155 @@
+    # in the /var/log/containers directory which includes the pod name and the Kubernetes
+    # container name:
+    #    synthetic-logger-0.25lps-pod_default-synth-lgr-997599971ee6366d4a5920d25b79286ad45ff37a74494f262e3bc98d909d0a7b.log
+    #    ->
+    #    /var/lib/docker/containers/997599971ee6366d4a5920d25b79286ad45ff37a74494f262e3bc98d909d0a7b/997599971ee6366d4a5920d25b79286ad45ff37a74494f262e3bc98d909d0a7b-json.log
+    # The /var/log directory on the host is mapped to the /var/log directory in the container
+    # running this instance of Fluentd and we end up collecting the file:
+    #   /var/log/containers/synthetic-logger-0.25lps-pod_default-synth-lgr-997599971ee6366d4a5920d25b79286ad45ff37a74494f262e3bc98d909d0a7b.log
+    # This results in the tag:
+    #  var.log.containers.synthetic-logger-0.25lps-pod_default-synth-lgr-997599971ee6366d4a5920d25b79286ad45ff37a74494f262e3bc98d909d0a7b.log
+    # The record reformer is used is discard the var.log.containers prefix and
+    # the Docker container ID suffix and "kubernetes." is pre-pended giving the tag:
+    #   kubernetes.synthetic-logger-0.25lps-pod_default-synth-lgr
+    # Tag is then parsed by google_cloud plugin and translated to the metadata,
+    # visible in the log viewer
+
+    # Example:
+    # {"log":"[info:2016-02-16T16:04:05.930-08:00] Some log text here\n","stream":"stdout","time":"2016-02-17T00:04:05.931087621Z"}
+    <source>
+      type tail
+      format json
+      time_key time
+      path /var/log/containers/*.log
+      pos_file /var/log/gcp-containers.log.pos
+      time_format %Y-%m-%dT%H:%M:%S.%N%Z
+      tag reform.*
+      read_from_head true
+    </source>
+
+    <filter reform.**>
+      type grep
+      <regexp>
+        key log
+        pattern \"component\":\"
+      </regexp>
+    </filter>
+
+    <filter reform.**>
+      type parser
+      format json
+      key_name log
+      time_format %Y-%m-%dT%H:%M:%S%Z
+    </filter>
+
+    <filter raw.kubernetes.**>
+      type stdout
+    </filter>
+
+    <match reform.**>
+      type record_reformer
+      enable_ruby true
+      tag raw.kubernetes.${tag_suffix[4].split('-')[0..-2].join('-')}
+    </match>
+
+    # Detect exceptions in the log output and forward them as one log entry.
+    <match raw.kubernetes.**>
+      @type copy
+
+      <store>
+        @type prometheus
+
+        <metric>
+          type counter
+          name logging_line_count
+          desc Total number of lines generated by application containers
+          <labels>
+            tag ${tag}
+          </labels>
+        </metric>
+      </store>
+      <store>
+        @type detect_exceptions
+
+        remove_tag_prefix raw
+        message log
+        stream stream
+        multiline_flush_interval 5
+        max_bytes 500000
+        max_lines 1000
+      </store>
+    </match>
+  output.conf: |-
+    <match kubernetes.**>
+      @type copy
+
+      <store>
+        @type google_cloud
```

Comment:
Does this work in build01 since we're on AWS?

Reply:
I have not gone that far. (Actually that is one of my questions. I do not see any info about GCE project and so on. I guess it uses the same one which the VMs are)

All i tried is
* privileged pods of a DS can run only on 3 masters
* they can access the audit-log files from the host